sql/src/main/java/org/apache/druid/sql/calcite/run/NativeQueryMaker.java

```diff
@@ -122,6 +125,24 @@ public boolean feature(QueryFeature feature)
         );
       }
     }
+    int numFilters = plannerContext.getPlannerConfig().getMaxNumericInFilters();
+    if (numFilters < 1) {
```

Comment:
Can this validation be moved somehow outside the `NativeQueryMaker`, since this is not specific to runQuery, but to when the `PlannerConfig` gets generated? There can be 2 cases to it: 
1. When overriding incorrectly in the specific query (i.e. `PlannerConfig#withOverrides` method)
2. When passing the incorrect value as `druid.sql.planner.maxNumericInFilter` -> Not sure how to achieve this since we can't validate @JsonProperty I think.
Let me know wdyt about moving this check elsewhere?

Reply:
@LakshSingla the check has been moved to `PlannerConfig`